lua/ttt2/extensions/sql.lua

```diff
@@ -0,0 +1,101 @@
+---
+-- sql extensions
+-- @author Histalek
+
+if SERVER then
+	AddCSLuaFile()
+end
+
+---
+-- Checks if the specified column exists in the specified table.
+-- @param string tableName The name of the table to search.
+-- @param string columnName The name of the column to check.
+-- @return boolean Returns true if the column exists in the table.
+-- @realm shared
+function sql.ColumnExists(tableName, columnName)
+	local result = sql.Query("PRAGMA table_info(" .. sql.SQLIdent(tableName) .. ")")
+
+	for i = 1, #result do
+		if result[i].name == columnName then
+			return true
+		end
+	end
+
+	return false
+end
+
+---
+-- Returns the primarykey column names of the specified table in order of their index.
+-- @param string tableName The name of the table to search.
+-- @return table|nil Returns a table of the primarykey columns.
+-- @realm shared
+function sql.GetPrimaryKey(tableName)
+	local result = sql.Query("PRAGMA table_info(" .. sql.SQLIdent(tableName) .. ")")
+	local primaryKeys = {}
+
+	for i = 1, #result do
+		if tonumber(result[i].pk) ~= 0 then
+			primaryKeys[pk] = result[i].name
+		end
+	end
+
+	return primaryKeys
+end
+
+---
+-- Returns the foreignkeys of the specified table.
+-- @param string tableName The name of the table to search.
+-- @return table|nil Returns a table of the foreignkey columns.
+-- @realm shared
+function sql.GetForeignKeys(tableName)
+	local result = sql.Query("PRAGMA foreign_key_list(" .. sql.SQLIdent(tableName) .. ")")
+	local foreignKeys = {}
+
+	if not result then return end
+
+	for i = 1, #result do
+		local ressultRow = result[i]
+		local id = tonumber(ressultRow.id)
+		local seq = tonumber(ressultRow.seq)
+		local foreignKeysRow = foreignKeys[id][seq]
+
+		foreignKeysRow.table = ressultRow.table
+		foreignKeysRow.from = ressultRow.from
+		foreignKeysRow.to = ressultRow.to
+	end
+
+	return foreignKeys
+end
+
+---
+-- Returns the column names of the specified table.
+-- @param string tableName The name of the table to search.
+-- @return table|nil Returns a table of the column names.
```

Comment:
nil is never returned

Reply:
well it does if you specify a table that does not exist =D
should i explicitly check if the table exists in here and return early?

Reply:
No it does not, it will always return an empty table

Reply:
fixed